src/main/java/states/ReadState.java

```diff
@@ -48,60 +52,129 @@ public ReadState(SocketAddress serverAddress, String filename, boolean isVerbose
 
 	@Override
 	public State execute() {
-		int expBlockNum = -1;
-		try {
-			// Set up + Send RRQ Message
+
+        expBlockNum = -1;
+        isConsentual = true;
+        try {
+            // Create the resource manager, handle IOException if failed to get resource directory
+
+            socket = new TFTPDatagramSocket();
```

Comment:
The socket is initialized in the constructor. This is done to run the mock integration tests properly

Reply:
That was a mistake on my part. Missed it during a rebase. Will fix.